e2e/terraform/provision-infra/provision-nomad/etc/nomad.d/client-linux-1.hcl

```diff
@@ -5,4 +5,8 @@ client {
   meta {
     "rack" = "r2"
   }
+
+  host_volume "shared_data" {
+    path = "/srv/data"
+  }
```

Comment:
I don't see anything that creates this dir, which needs to happen before agent start.  unless... you could use a DHV?

Reply:
It's on all the client hosts already, created in the packer image at [`setup.sh#L23`](https://github.com/hashicorp/nomad/blob/7a051991bd1791d3ad7b0c41600672a5956eb7f0/e2e/terraform/packer/ubuntu-jammy-amd64/setup.sh#L23). One of the clients already has it in the config, so this is just giving us more potential clients to deploy the NFS server too.

Unfortunately, we can't use a DHV in upgrade testing because the workloads need to be running before the upgrade, and 1.8.x LTS won't have DHV.